code/game/objects/effects/particles_system.dm

```diff
@@ -111,10 +131,48 @@
 	icon_state = "blank"
 	var/atom/main_holder
 
+/obj/abstract/particles_holder/proc/special_setup(var/particle_string)
+	switch(particle_string)
```

Comment:
Particles don't have layers, planes, blend_mode or appearance_flags. But they do inherit those of the holder they're part of. This entire system is basically to go around Byond limitations.

The alternative would be to create a dedicated holder subtype for every particle subtype which I kinda wished to avoid. Do you think it's better.

Reply:
Couldnt you do this (I can't think of a better name than contains_attributes, I don't like that)
```
/particle/contains_attributes
	holder_appearance_flags = null
	holder_blend_mode = null
	holder_plane = null
	
/particle/contains_attributes/candle
	holder_appearance_flags = RESET_COLOR
	holder_blend_mode = BLEND_ADD
	holder_plane = ABOVE_LIGHTING_PLANE
	
/obj/abstract/particles_holder/proc/special_setup(var/particle)
	/particle/contains_attributes/conversion = particle
	if(istype(conversion))
	  if(conversion.holder_appearance_flags != null)
	     src.appearance_flags = conversion.holder_appearance_flags
	  and so on for the other two
```

Reply:
Would be nice if byond was complex to allow more than simple inheritance chains but this should work and be extendible to any particle use in the future.

Reply:
Actually I realized I could just add vars to the base /particle type so that works